.github/workflows/ci.yml

```diff
@@ -0,0 +1,86 @@
+name: CI
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest 
+
+    steps:
+      - uses: actions/checkout@v2
+      
+    #to speed up the execution time of Github Actions
+      - name: Cache multiple paths  
+        uses: actions/cache@v2
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: |-
+            vendor
+            node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+                ${{ runner.os }}-yarn-
+     
+      - uses: getong/mariadb-action@v1.1  
+        with:
+          host port: 3306 
+          mariadb version: '10.5' 
+          mysql database: 'test' 
+      
+      - uses: ruby/setup-ruby@v1    
+        with:
+           ruby-version: 2.7.1
+           bundler-cache: true
+
+           
+      - name: Setup Node
+        uses: actions/setup-node@v1
+               
+      - run: 'export PATH=$PATH:`yarn global bin`'
+      
+      - run: export DB=test
+      
+      - run: export SENDER_EMAIL_ADDRESS='sender@wikiedu.org'
+      
+      - name: Installing Packages 
+        run: |
+            yarn install
+            gem install bundler
+            gem install rubocop
+            gem install rubocop-rspec-focused
+            bundle config set path 'vendor'
+            bundle install --jobs=3 --retry=3
+            
+      - run: mkdir tmp -m 777
+      - run: cp config/database.example.yml config/database.yml
+      - run: cp config/application.example.yml config/application.yml
+
+      - name:  Creating Test Database 
+        env:
+          RAILS_ENV: test
+          DATABASE_PORT: 3306
+        run: |
+            bin/rails db:create
+            bin/rails db:migrate RAILS_ENV=test
+      
+      - name: Running tests   
```

Comment:
It looks like this isn't running the the JS tests, which are run via `yarn test`.

Ideally, in terms of the output and how a test run shows up, we want to know which of the test suites and checks passed and which failed. Those consist of: JS tests (`yarn test`), Ruby tests (`bundle exec rspec ...`), Ruby linting (`bundle exec rubocop`) and JS linting (`yarn lint-non-build`). Maybe that means putting each one in a separate step?

Reply:
oh no i forget that too add in my final commit i will be putting it under the test itself and will definitely work fine or will sperate few of them under linting  to make it look cleaner.